### ISSUE TITLE
What is target in the losses function? 

### ISSUE THREAD

**Issue by cam1681** [lululxvi/deepxde#358]
Hello, thanks for your wonderful code, especially the efforts for PyTorch!

What is the variable "target"  in the "losses" function? I tested the Possion_Lshape.py and print the "targets" in the losses function and I found that it is always None. So, how does the target plays in the losses?  **If I want the residual not the loss, i.e., the vector of [residualPDE, boundary_difference], how can I modify the code?** 

Thanks again!

**Comment by cam1681** [comment]
It seems that the answer is buried in the data/pde.py.  error_f and error in the boundary give the residual vector and the difference on the boundary.

**Comment by lululxvi** [comment]
The points on the boundary are used twice: once of the PDE residual, and one for the boundary loss.

**Comment by cam1681** [comment]
> The points on the boundary are used twice: once of the PDE residual, and one for the boundary loss.

Thanks! Another question, are the points for training randomly chosen at each iteration when using L-BFGS or Adam? If not, how could I modify the code to randomly sampling training data in the domain?

**Comment by lululxvi** [comment]
You can use `dde.callbacks.PDEResidualResampler` for point resampling in each every iterations, see https://github.com/lululxvi/deepxde/blob/master/examples/diffusion_1d_resample.py

**Comment by cam1681** [comment]
Thank you very much for your reply, I have got what I want.
